cmd/noms/diff/diff.go

```diff
@@ -184,40 +207,33 @@ func diffOrdered(w diffWriter, p types.Path, lf lineFunc, ppf pathPartFunc, df d
 		close(changeChan)
 	}()
 
-	wroteHdr := false
-
 	for change := range changeChan {
-		if err != nil {
+		if stop = stopSent(sc); stop {
```

Comment:
I prefer the way that the ordered diff does this. Remove this code, then instead of
```
dc <- Difference{...}
```
have
```
if !sendDiff(dc, sc, Difference{...}) {
  break Range
}
```
(you'll need a Range label on the outer `for` to be able to break out of the switch... or use some variable I suppose)

Even better: if you leave that struct in place that bundles state, you can put the `sendDiff` function on that, and not need to pass in those 2 parameters separately.